### ISSUE TITLE
row editor should resize when the window resizes (and show scrollbar)

### ISSUE THREAD

**Issue by bartvde** [MapStory/mapstory-meta#877]
http://help.mapstory.org/question/198/annotation-window-distorted/?comment=201#comment-201

**Comment by bartvde** [comment]
This is actually caused by the override we had to support row editor on the last row:

```
Ext.override(Ext.grid.GridView, {
    getEditorParent: function() {
        return document.body;
    }       
});
```

**Comment by bartvde** [comment]
hmm it seems that the row editor ux from Sencha is not really prepared for this use case, seeing if there is still something we can do about this one

**Comment by bartvde** [comment]
I can't really get an ideal solution for this. The solution I had works fine for the first resize, but then the next row will be messed up:

```
diff --git a/src/geonode-client/app/static/script/app/mapstory/Notes.js b/src/geonode-client/app/static/script/app/mapstory/Notes.js
index 4c17db1..2a47a93 100644
--- a/src/geonode-client/app/static/script/app/mapstory/Notes.js
+++ b/src/geonode-client/app/static/script/app/mapstory/Notes.js
@@ -431,6 +431,16 @@ mapstory.plugins.NotesManager = Ext.extend(gxp.plugins.Tool, {
             }
             this.pos_ = [x, y];
         }, this);
+        win.on('resize', function() {
+            var editor = output.plugins[0];
+            if (editor && editor.rendered) {
+                editor.verifyLayout();
+                var w = editor.getWidth();
+                editor.setWidth(win.getWidth()-15);
+                editor.body.setWidth(w);
+                Ext.get(editor.body.dom.firstChild).setWidth(w);
+            }
+        });
         return win;
     },
```

not sure how important this one is, but I'm inclined to leave it.

**Comment by bartvde** [comment]
okay I decided to remove the override that made the editor parent the body. This doesn't mean we get a scrollbar on the row editor when we make the window too small, but the row editor will never be bigger than the window. Since we changed the annotations window to be resizable, the last row can now be edited though this requires the user to scroll a little bit. I think this is as good as it gets for now unfortunately.

https://github.com/opengeo/geonode/commit/d817ba75e9ccfaa7f1d57c7e3dc2c5229aff8df4
